### ISSUE TITLE
Using micrometer-registry-statsd doesn't report TimeUnit to DataDog

### ISSUE THREAD

**Issue by ayeletmorris** [micrometer-metrics/micrometer#2528]
Hi, I'm using micrometer-registry-statsd with @Timed to report time series data to DataDog, and all of my metrics have "none" in the time unit section in the metrics summary page.

I am trying to understand what is the TimeUnit that is being used and to even define it manually and I saw in your code that the timers use Nanoseconds to save the data, but when I defined the metrics time unit as nanoseconds the data makes no sense, I am trying to dig deeper and I saw that in the StatsdMeterRegistry class the "getBaseTimeUnit" method returns Milliseconds, so now I'm confused...

Basically my questions are:
1. What time unit does the timed annotation (timers) report to dataDog?
2. Why doesn't it show the supposedly reported time unit int he dataDog metric?

**Comment by shakuzen** [comment]
> I saw in your code that the timers use Nanoseconds to save the data

Nanoseconds is an internal representation. Depending on the registry, timings will be converted to the base unit for that implementation, according to what is idiomatic or required for that backend.

>     1. What time unit does the timed annotation (timers) report to dataDog?

You're using the statsd registry, and [the statsd standard for timing is milliseconds](https://github.com/statsd/statsd/blob/master/docs/metric_types.md#timing). So we report it in milliseconds.

> Why doesn't it show the supposedly reported time unit int he dataDog metric?

That's really a question for the maintainers of DogStatsD. It seems like they could include such metadata but aren't currently, as far as I can tell.
If you were using the DatadogMeterRegistry (and you configure an application key) instead of statsd, we do report the time unit metadata for metrics since we directly report them to Datadog.
